### ISSUE TITLE
matnwb cannot read output NWB file

### ISSUE THREAD

**Issue by CBroz1** [datajoint/element-array-ephys#81]
A user reported difficulty reading an nwb file produced by the export function.

<img width="633" alt="Screenshot 2022-06-27 at 18 04 43" src="https://user-images.githubusercontent.com/9404707/176010235-a687ca38-2e92-4ec8-985a-4d9af26f84a9.png">

@bendichter - Should files exported here be readable with `matnwb` ?

**Comment by bendichter** [comment]
@CBroz1 yes it should be. Would you be able to send me this file? @lawrence-mbf, do you have any idea what might be going on here?

**Comment by CBroz1** [comment]
> @CBroz1 yes it should be. Would you be able to send me this file? 

Of course. I'll send over slack. Thanks for looking into it!

**Comment by lawrence-mbf** [comment]
@CBroz1 This unfortunately is a really old issue where MatNWB doesn't support anonymous type definitions: https://github.com/NeurodataWithoutBorders/matnwb/issues/43

Thankfully, I think this is localised in the `waveform_mean` VectorData object as we haven't seen the same elsewhere. A workaround could just be to make a `waveform_mean_unit` vectordata object if you really need it in MatNWB.

**Comment by CBroz1** [comment]
Thanks @lawrence-mbf for your insights!

@kabilar - I'm thinking we should make that edit so resulting files are more accessible. What do you think?

**Comment by kabilar** [comment]
Thanks all.  Yes, let's make the update.  I couldn't find documentation on `waveform_mean_unit` so not sure how we would make this update.

**Comment by lawrence-mbf** [comment]
@kabilar @CBroz1 It would really just be repeating that `unit` value as a vector column with the same number of rows. It is an ugly duplication of code and may one day throw warnings when validating but it will guarantee MatNWB support for now.
